### PR TITLE
integrate game-activity

### PR DESCRIPTION
**Pull request by bofeng-song**

### CODE REVIEW

File: native/cocos/platform/android/AndroidPlatform.cpp [hunk 1]
```diff
@@ -25,79 +25,460 @@
...
 #include "modules/System.h"
+#include "java/jni/log.h"
+#include "java/jni/JniHelper.h"
 
-namespace cc {
-AndroidPlatform::AndroidPlatform() {
-    _jniNativeGlue = JNI_NATIVE_GLUE();
-}
+#include "game-activity/GameActivity.cpp"
+#include "game-text-input/GameTextInput.cpp"
```
Comment (dumganhar) [inline native/cocos/platform/android/AndroidPlatform.cpp]: `cpp`, `c` files need  to be marked as source file in CMakeLists.txt.

File: native/cocos/platform/android/AndroidPlatform.cpp [hunk 2]
```diff
@@ -25,79 +25,460 @@
...
+        int32_t actionCode;
+    };
+
+    static const InputAction PADDLEBOAT_ACTIONS[INPUT_ACTION_COUNT] = {
+            {PADDLEBOAT_BUTTON_A,          static_cast<int>(KeyCode::ENTER)},
+            {PADDLEBOAT_BUTTON_B,          static_cast<int>(KeyCode::ESCAPE)},
+            {PADDLEBOAT_BUTTON_DPAD_UP,    1003},
+            {PADDLEBOAT_BUTTON_DPAD_LEFT,  1000},
+            {PADDLEBOAT_BUTTON_DPAD_DOWN,  1004},
+            {PADDLEBOAT_BUTTON_DPAD_RIGHT, 1001}
```
Comment (minggo) [inline native/cocos/platform/android/AndroidPlatform.cpp]: Add these magic values to KeyCode?

File: native/cocos/platform/android/AndroidPlatform.cpp [hunk 3]
```diff
@@ -25,79 +25,460 @@
...
+
+namespace cc {
+
+    struct cc::TouchEvent touchEvent;
+    struct cc::KeyboardEvent keyboardEvent;
+    static uint32_t prevButtonsDown = 0;
+
+    struct InputAction {
+        uint32_t buttonMask;
+        int32_t actionCode;
```
Comment (minggo) [inline native/cocos/platform/android/AndroidPlatform.cpp]: default values

File: native/cocos/platform/android/AndroidPlatform.cpp [hunk 4]
```diff
@@ -25,79 +25,460 @@
...
-    _jniNativeGlue->setRunning(true);
-    while (_jniNativeGlue->isRunning()) {
-        pollEvent();
-        NativeWindowType *wndHandle = _jniNativeGlue->getWindowHandler();
-        if (wndHandle != nullptr) {
-            break;
+#define ABORT_GAME { ALOGE("*** GAME ABORTING."); *((volatile char*)0) = 'a'; }
+
+#define MY_ASSERT(cond) { if (!(cond)) { ALOGE("ASSERTION FAILED: %s", #cond); \
+   ABORT_GAME; } }
```
Comment (minggo) [inline native/cocos/platform/android/AndroidPlatform.cpp]: Why not use CC_ASSERT?

File: native/cocos/platform/android/AndroidPlatform.cpp [hunk 5]
```diff
@@ -25,79 +25,460 @@
...
+                    touchEvent.type = cc::TouchEvent::Type::ENDED;
+                } else if (actionMasked == AMOTION_EVENT_ACTION_CANCEL) {
+                    touchEvent.type = cc::TouchEvent::Type::CANCELLED;
+                } else if (actionMasked == AMOTION_EVENT_ACTION_MOVE) {
+                    touchEvent.type = cc::TouchEvent::Type::MOVED;
+                } else {
+                    return false;
+                }
+
+                for (int i = 0; i < motionEvent->pointerCount; i++) {
```
Comment (minggo) [inline native/cocos/platform/android/AndroidPlatform.cpp]: Use for(a : b)?

File: native/cocos/platform/android/AndroidPlatform.cpp [hunk 6]
```diff
@@ -25,79 +25,460 @@
...
+            bool controllerEvent = false;
+
+            // Swap input buffers so we don't miss any events while processing inputBuffer.
+            android_input_buffer *inputBuffer = android_app_swap_input_buffers(
+                    _androidPlatform->_app);
+            // Early exit if no events.
+            if (inputBuffer == nullptr) return;
+
+            if (inputBuffer->keyEventsCount != 0) {
+                for (uint64_t i = 0; i < inputBuffer->keyEventsCount; ++i) {
```
Comment (minggo) [inline native/cocos/platform/android/AndroidPlatform.cpp]: Use for(a : b)?

File: native/cocos/platform/android/AndroidPlatform.cpp (hunk 4 above)
Comment (bofeng-song) [reply native/cocos/platform/android/AndroidPlatform.cpp]: expected to crash, while cond is false

Comment (minggo) [reply native/cocos/platform/android/AndroidPlatform.cpp]: Also in release mode?

Comment (bofeng-song) [reply native/cocos/platform/android/AndroidPlatform.cpp]: right, if wrong logic occur, crash at right position

File: native/cocos/platform/android/AndroidPlatform.cpp (hunk 5 above)
Comment (bofeng-song) [reply native/cocos/platform/android/AndroidPlatform.cpp]: no viable 'begin' function available

File: native/cocos/platform/android/AndroidPlatform.cpp (hunk 6 above)
Comment (bofeng-song) [reply native/cocos/platform/android/AndroidPlatform.cpp]: no viable 'begin' function available

File: native/cocos/platform/android/AndroidPlatform.cpp (hunk 1 above)
Comment (bofeng-song) [reply native/cocos/platform/android/AndroidPlatform.cpp]: use GameActivityInclude.cpp to include these files, CMakeLists.txt can not get the absoulte path of GameActivity.cpp

File: native/cocos/platform/android/AndroidPlatform.cpp (hunk 4 above)
Comment (minggo) [reply native/cocos/platform/android/AndroidPlatform.cpp]: Got it. It is better not to use ASSERT, as it is confused with `assert`. And it is better not use `MY`, use a more meaningful name please.

Comment (minggo) [reply native/cocos/platform/android/AndroidPlatform.cpp]: How about LOG_AND_ABORT?

Comment (minggo) [reply native/cocos/platform/android/AndroidPlatform.cpp]: Or ABORT_IF.

**Pull request closed without merging**